app/src/main/java/emu/skyline/EmulationActivity.kt

```diff
@@ -211,17 +211,27 @@ class EmulationActivity : AppCompatActivity(), SurfaceHolder.Callback, View.OnTo
         shouldFinish = true
         returnToMain = intent.getBooleanExtra(ReturnToMainTag, false)
 
-        val rom = intent.data!!
-        val romType = getRomFormat(rom, contentResolver).ordinal
-        val romFd = contentResolver.openFileDescriptor(rom, "r")!!
-        val preferenceFd = ParcelFileDescriptor.open(File("${applicationInfo.dataDir}/shared_prefs/${applicationInfo.packageName}_preferences.xml"), ParcelFileDescriptor.MODE_READ_WRITE)
+        intent.data?.let { rom ->
+            val romType = getRomFormat(rom, contentResolver).ordinal
+            val romFd = contentResolver.openFileDescriptor(rom, "r")
+            val preferenceFd = ParcelFileDescriptor.open(File("${applicationInfo.dataDir}/shared_prefs/${applicationInfo.packageName}_preferences.xml"), ParcelFileDescriptor.MODE_READ_WRITE)
+
+            emulationThread = Thread {
+                executeApplication(
+                    romUri = rom.toString(),
+                    romType = romType,
+                    romFd = romFd?.detachFd()?:0,
+                    preferenceFd = preferenceFd.detachFd(),
+                    language = settings.systemLanguage,
+                    appFilesPath = applicationContext.filesDir.canonicalPath + "/",
+                    nativeLibraryPath = applicationInfo.nativeLibraryDir + "/",
+                    assetManager = assets
+                )
+                returnFromEmulation()
+            }
 
-        emulationThread = Thread {
-            executeApplication(rom.toString(), romType, romFd.detachFd(), preferenceFd.detachFd(), settings.systemLanguage, applicationContext.filesDir.canonicalPath + "/", applicationInfo.nativeLibraryDir + "/", assets)
-            returnFromEmulation()
+            emulationThread?.start()
         }
-
-        emulationThread!!.start()
```

Comment:
This is code that start the emulation thread, which in turns runs the game.
Here you just replaced the not-null assertion operator with a safe calls, but without providing a viable fallback path in case the value is null. The app would do nothing in this case and just hang, this is even worse than crashing.

Reply:
It might solve the concern to crash more gracefully in such instances, since a fatal crash looks unplanned and it sounds like a crash is an expected outcome.

Reply:
> It might solve the concern to crash more gracefully in such instances, since a fatal crash looks unplanned and it sounds like a crash is an expected outcome.

It wouldn't be more gracefully, because yes, if this goes wrong, a crash is the expected behavior, and this doesn't even throw a more meaningful exception so it's probably less useful for debugging than a NPE

Reply:
@MCredstoner2004 I think you misunderstood. I was suggesting that the crash is *handled* more gracefully, as it is currently not being handled at all. That doesn't mean *preventing* it, as the PR is attempting.

Reply:
I see what you mean, but there's really nothing to handle here. If those objects are null where the `!!` are placed, the app should crash as that means unrecoverable failure. Although, the circumstances when this would happen are very rare.